ansible/roles/dashmate/tasks/main.yml

```diff
@@ -148,30 +149,113 @@
-  ignore_errors: true
+  register: dashmate_status
+
+- name: Get service status
+  ansible.builtin.set_fact:
+    core_docker_status: "{{ dashmate_status.stdout | from_json | json_query('core.dockerStatus') }}"
+    tenderdash_docker_status: "{{ dashmate_status.stdout | from_json | json_query('platform.tenderdash.dockerStatus') }}"
+    core_images_updated: >-
+      {{
+        dashmate_update.stdout |
+        from_json |
+        json_query('[?name==`core`||name==`sentinel`]') |
+        sort(attribute='updated', reverse=true) |
+        first() |
+        json_query('updated')
+      }}
+    platform_images_updated: >-
+      {{
+        dashmate_update.stdout |
+        from_json |
+        json_query('[?name!=`core`&&name!=`sentinel`]') |
+        sort(attribute='updated', reverse=true) |
+        first() |
+        json_query('updated')
+      }}
+    dashmate_package_changed: "{{ dashmate_install_result.changed }}"
+    dashmate_config_changed: "{{ template_result.changed }}"
+
+- name: Print status
+  ansible.builtin.debug:
+    msg: "\n
+Core Status: {{ core_docker_status }}\n
+Tenderdash Status: {{ tenderdash_docker_status }}\n
+Core Images Updated: {{ core_images_updated }}\n
+Platform Images Updated: {{ platform_images_updated }}\n
+Dashmate Package Updated: {{ dashmate_package_changed }}\n
+Dashmate Config Changed: {{ dashmate_config_changed }}
+"
+
+# Start logic:
+# start ALL when (core is down and platform is down)
+# start platform when (core is up and platform is down)
 
-# Start if not already running
+# Start all if everything is stopped (not started yet)
 - name: Start dashmate services
   ansible.builtin.command: "{{ dashmate_cmd }} start --verbose"
   become: true
   become_user: dashmate
   args:
     chdir: '{{ dashmate_cwd }}'
   register: dashmate_start
-  when: "dashmate_restart.rc != 0"
+  when:
+    - core_docker_status == 'not_started'
+    - tenderdash_docker_status == 'not_started'
   changed_when: dashmate_start.rc == 0
 
+# Start platform in case core is running
+- name: Start dashmate platform services
+  ansible.builtin.command: "{{ dashmate_cmd }} start --platform --verbose"
+  become: true
+  become_user: dashmate
+  args:
+    chdir: '{{ dashmate_cwd }}'
+  register: dashmate_start_platform
+  when:
+    - core_docker_status == 'running'
+    - tenderdash_docker_status == 'not_started'
+  changed_when: dashmate_start_platform.rc == 0
+
+# Restart logic
+# if any of platform images were updated => restart platform
+# if core or sentinel images updated => restart all
+# dashmate version is changed => restart all
+# dashmate config is changed => restart all
+
+- name: Restart dashmate platform services
+  ansible.builtin.command: "{{ dashmate_cmd }} restart --platform --verbose"
+  become: true
+  become_user: dashmate
+  args:
+    chdir: '{{ dashmate_cwd }}'
+  register: dashmate_restart
+  when:
+    - core_docker_status == 'running' and tenderdash_docker_status == 'running'
```

Comment:
Are they working as AND or OR?

Reply:
As AND. We only want to restart if all services are currently running